web-client/src/presenter/actions/handleIdleLogoutAction.ts

```diff
@@ -7,20 +7,25 @@ export const handleIdleLogoutAction = ({ get, path, store }: ActionProps) => {
   const idleLogoutState = get(state.idleLogoutState);
   const isUploading = get(state.fileUploadProgress.isUploading);
   const userIsLoggedIn = !!get(state.token);
+  const clientNeedsToRefresh = get(state.clientNeedsToRefresh);
 
-  if (
-    userIsLoggedIn &&
-    !isUploading &&
-    idleLogoutState.state === IDLE_LOGOUT_STATES.INITIAL
-  ) {
+  // Short-circuit here to prevent showing the "Are you still there?" modal, etc. when inappropriate
+  if (!userIsLoggedIn || clientNeedsToRefresh) {
+    return path.continue();
+  }
+
+  if (!isUploading && idleLogoutState.state === IDLE_LOGOUT_STATES.INITIAL) {
```

Comment:
Looking at the short history of this file, it appears this `!userIsLoggedIn` check only being in the initial `if` block was an unintentional artifact of a refactor. Since this was an easy mistake to make--the nested `if/else if` makes this logic difficult to understand at a glance--I've pulled each condition out into a separate `if/return` block.